[dagster-airflow] migration limitations (#12124)
dagster-io/dagster: docs/content/integrations/airflow/migrating-to-dagster.mdx

- **Using [Airflow connections](https://airflow.apache.org/docs/apache-airflow/stable/howto/connection.html) or [environment variables](/dagster-cloud/developing-testing/environment-variables-and-secrets)** are the easiest methods for defining permissions.

- **If your Airflow DAGs used [`KubernetesPodOperators`](https://airflow.apache.org/docs/apache-airflow-providers-cncf-kubernetes/stable/operators.html)**, it's possible that you used the `in_cluster` config or loading a `kube_config` file. When migrating, we recommend switching to [using connections with a `kube_config` JSON blob](https://airflow.apache.org/docs/apache-airflow-providers-cncf-kubernetes/stable/connections/kubernetes.html) to make things easier.

> ## Limitations

Reviewer: ```suggestion
---

## Limitations
```

H2 headings should be preceded by a horizontal rule

> Currently there are a few limitions of converting airflow DAGs into Dagster Jobs/Assets/Schedules:

Reviewer: ```suggestion
There are a few limitations to converting Airflow DAGs into Dagster Jobs, Assets, and Schedules. The following features are currently unsupported:
```

Some small grammar tweaks. Also, setting this up to cut down on some wordiness in the individual list items.

> - retry-from-failure in dagster is unsupported
> - `prev_execution_date` in airflow templated dags is unsupported
> - `SubDAGOperators` are unsupported
> - Airflow Datasets are unsupported

Reviewer: ```suggestion
- `retry-from-failure` in Dagster
- Using `prev_execution_date` in Airflow-templated DAGs
- `SubDAGOperators`
- Airflow Datasets
```

> Most of the current limitations are due to the fact that a locally scoped ephemeral airflow db is created for each run, in the future it will be possible to share this state accross runs.

Reviewer: ```suggestion
The majority of these limitations are due to a locally-scoped, ephemeral Airflow database being created for each run. In future Dagster releases, it will be possible to share this state across runs.
```

Suggesting changes for grammar and clarity

> Most of the current limitations are due the locally scoped ephemeral airflow db that is created for each run. If your DAGs implicitly rely on airflow database state across DagRuns then the current migration tooling will not work for you. We are working on extending support for these features so if you're interested in them let us know in the #dagster-airflow slack!

Reviewer: ```suggestion
Most of the current limitations are due to a locally-scoped, ephemeral Airflow database being created for each run. If your DAGs implicitly rely on Airflow database state across `DagRuns`, then the current migration tooling won't work for you.

Note that we are working on extending support for these features. If interested in them, let us know in the #dagster-airflow channel of the Dagster Slack.
```

Suggestions for grammar and clarity/readability
